packages/@aws-cdk/aws-eks/lib/cluster.ts

```diff
@@ -679,6 +680,19 @@ export class Cluster extends Resource implements ICluster {
     return this._clusterResource.getCreationRoleArn(assumedBy);
   }
 
+  /**
+   * Returns the custom resource provider for kubectl-related resources.
+   * @internal
+   */
+  public get _kubectlProvider(): KubectlProvider {
+    if (!this._clusterResource) {
+      throw new Error('Unable to perform this operation since kubectl is not enabled for this cluster');
+    }
```

Comment:
is there already an existing unit test that covers this?

minor: also maybe simplify error message. since it's internal I think it reads easier as `kubectl is not enabled for this cluster. <how to reconcile if possible to fix error>`

Reply:
There is a [test](https://github.com/aws/aws-cdk/blob/397752cc99d4a573bf5d75fc7ea3ef8d84613927/packages/%40aws-cdk/aws-eks/test/test.cluster.ts#L327) that covers this use case. The error indicates that kubectl is not enabled and that's the possible fix.